src/components/cylc/gscan/GScan.vue

```diff
@@ -102,79 +96,45 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
         </v-menu>
       </div>
       <!-- data -->
-      <div
-        v-if="!isLoading"
-        class="c-gscan-workflows flex-grow-1"
-      >
-        <tree
-          :filterable="false"
-          :expand-collapse-toggle="false"
-          :workflows="workflows"
-          :stopOn="['workflow']"
-          :autoExpandTypes="['workflow-part', 'workflow']"
-          class="c-gscan-workflow ma-0 pa-0"
-          ref="tree"
-        >
+      <div v-if="!isLoading" class="c-gscan-workflows flex-grow-1">
+        <tree :filterable="false" :expand-collapse-toggle="false" :workflows="workflows" :stopOn="['workflow']" :autoExpandTypes="['workflow-part', 'workflow']" class="c-gscan-workflow ma-0 pa-0" ref="tree">
+          <!-- this overwrites the node slot in the tree-item template, these are ALWAYS inserted where the node slot is regardless of its current value -->
           <template v-slot:node="scope">
-            <workflow-icon
-              class="mr-2"
-              v-if="scope.node.type === 'workflow'"
-              :status="scope.node.node.status"
-              v-cylc-object="scope.node"
-            />
-            <v-list-item
-              :to="workflowLink(scope.node)"
-            >
+
+            <workflow-icon class="mr-2" v-if="scope.node.type === 'workflow' || !checkForBranchingLineage(scope.node)" :status="getLastDescendent(scope.node).node.status || 'waiting'" v-cylc-object="getLastDescendent(scope.node)" />
```

Comment:
`checkForBranchingLineage` and `getLastDescendent` seem to be used multiple times which means there will be duplicate computation. It would be worth using computed properties to cache these values to avoid unnecessary computation.

Reply:
'The auto-collapsing works well for the examples 👍, although for some strange reason, when I tried it with a workflow named generic/run1 it didn't work and the workflow icon broke?'

I actually worked out why this was happening after I left the office last night, I will fix it today